graveler/sstable/cache.go

```diff
@@ -43,22 +53,27 @@ func (i *item) Close() error {
 	return i.r.Close()
 }
 
-func NewCache(p lru.ParamsWithDisposal, fs pyramid.FS, readerOptions sstable.ReaderOptions) cache {
-	return NewCacheWithOpener(p, func(namespace string, id committed.ID) (Item, error) {
-		file, err := fs.Open(namespace, string(id))
-		if err != nil {
-			return nil, fmt.Errorf("fetch %s from next tier: %w", id, err)
-		}
-
-		reader, err := sstable.NewReader(file, readerOptions)
-		if err != nil {
-			return nil, fmt.Errorf("open SSTable %s: %w", id, err)
-		}
-		return &item{reader}, nil
-	})
+func NewCache(p lru.ParamsWithDisposal, fs pyramid.FS, readerOptions sstable.ReaderOptions) Cache {
+	return NewCacheWithOpener(p,
+		func(namespace string, id string) (Item, error) {
+			file, err := fs.Open(namespace, string(id))
+			if err != nil {
+				return nil, fmt.Errorf("fetch %s from next tier: %w", id, err)
+			}
+
+			reader, err := sstable.NewReader(file, readerOptions)
+			if err != nil {
+				return nil, fmt.Errorf("open SSTable %s: %w", id, err)
+			}
+			return &item{reader}, nil
+		},
+		func(namespace string, id string) (bool, error) {
+			return fs.Exists(namespace, id)
+		},
+	)
 }
 
-func NewCacheWithOpener(p lru.ParamsWithDisposal, open opener) cache {
+func NewCacheWithOpener(p lru.ParamsWithDisposal, open opener, exists existser) Cache {
```

Comment:
1. You don't set the `exists` arg.
2. What's the added value of passing a func and calling it later instead of just declaring and using it when needed?